sdk/dotnet/Pulumi/Core/Output.cs

```diff
@@ -131,6 +131,12 @@ public static Output<T> Create(Task<T> value)
         internal static Output<T> CreateSecret(Task<T> value)
             => Create(value, isSecret: true);
 
+        internal async Task<OutputData<T>> ChangeSecretness(Task<bool> isSecret)
```

Comment:
`Task<bool>` is fishy - I'll change it to `bool` or `Input<bool>`, but let's first figure out the public API of references

Reply:
this is fine currently (as it's not a public member).

Reply:
this is defined on Output but doesn't return an output.  i would change this to: `internal Output<T> WithIsSecret(Task<bool> isSecret)`

Reply:
done, thanks!